cases/interactive-flows.optional.test.js

```diff
@@ -0,0 +1,269 @@
+      let respJSON;
+      let timePassed = 0;
+      while (timePassed < timeAllowed) {
+        await new Promise((resolve) => {
+          setTimeout(resolve, interval);
+        });
+        timePassed += interval;
+        let transactionResp = await fetch(
+          toml.TRANSFER_SERVER + `/transaction?id=${transactionId}`,
+          {
+            headers: {
+              Authorization: `Bearer ${jwt}`,
+            },
+          },
+        );
+        respJSON = await transactionResp.json();
+        if (respJSON.transaction.status === "completed") {
+          break;
+        }
+      }
+      resolve(respJSON.transaction);
+    });
+    if (transactionJSON.status === "completed") {
+      resolve(transactionJSON);
+    } else {
+      reject("Timed out while waiting for transaction to complete!");
+    }
+  });
+}
+
+beforeAll(async () => {
+  await fetch(`https://friendbot.stellar.org?addr=${keyPair.publicKey()}`);
+  try {
+    toml = await getTomlFile(url);
+  } catch (e) {
+    throw "Invalid TOML formatting";
+  }
+
+  const infoResponse = await fetch(toml.TRANSFER_SERVER + "/info", {
+    headers: {
+      Origin: "https://www.website.com",
+    },
+  });
+  infoJSON = await infoResponse.json();
+  const currencies = Object.keys(infoJSON.withdraw);
+  // Note that we're only testing the first asset found to be enabled
+  enabledCurrency = currencies.find(
+    (currency) => infoJSON.withdraw[currency].enabled,
+  );
+  jwt = await getSep10Token(url, keyPair);
+
+  // Get info for interactive tests
+  currencyInfo = toml.CURRENCIES.find((obj) => obj.code === enabledCurrency);
+  issuerAccount = currencyInfo.issuer;
+  asset = new StellarSDK.Asset(enabledCurrency, issuerAccount);
+  account = await server.loadAccount(keyPair.publicKey());
+
+  // Establish trustline
+  let transaction = new StellarSDK.TransactionBuilder(account, {
+    fee: StellarSDK.BASE_FEE,
+    networkPassphrase: "Test SDF Network ; September 2015",
+  })
+    .addOperation(
+      StellarSDK.Operation.changeTrust({
+        asset: asset,
+        source: keyPair.publicKey(),
+      }),
+    )
+    .setTimeout(30)
+    .build();
+  transaction.sign(keyPair);
+
+  await server.submitTransaction(transaction);
+});
+
+let depositTransactionJSON = null;
+describe("Deposit Flow", () => {
+  let transactionId;
+  let transactionJSON;
+  it("can complete the interactive flow", async () => {
+    transactionId = await doInteractiveFlow({
+      currency: enabledCurrency,
+      account: keyPair.publicKey(),
+      jwt: jwt,
+      toml: toml,
+      isDeposit: true,
+    });
+    let transactionRespJSON = await getTransactionBy({
+      iden: "id",
+      value: transactionId,
+      toml: toml,
+      jwt: jwt,
+    });
+    transactionJSON = transactionRespJSON.transaction;
+  });
+
+  it("pending transactions are completed on testnet", async () => {
+    // Wait for interactive flow to complete
+    await waitUntilTruthy({ val: transactionJSON }, 15000, 1000);
```

Comment:
Add some commenting about how this 'force serialized tests' scheme works for the next person